### PR TITLE
chore: enable cache for npm on GitHub Actions

### PR DESCRIPTION
**Pull request by sosukesuzuki**
ref: https://github.com/actions/setup-node#caching-packages-dependencies

Now we can enable cache for npm dependencies with `actions/setup-node@v2`

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,34 +15,34 @@ jobs:
...
-        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-      run: npx codecov
 
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
```
Comment (sosukesuzuki) [inline .github/workflows/test.yml:25]: The only real change is in this line. Other lines were changed by Prettier via my text editor formatter.

**Review by koba04 — APPROVED** [review]
👍 Great 👏

**Comment by koba04** [comment]
@sosukesuzuki Should we use `npm install` instead of `npm ci` to re-use the cache?

**Comment by sosukesuzuki** [comment]
@koba04 No, maybe we can use `npm ci`. Why `npm install` is used in README of [actions/setup-node](https://github.com/actions/setup-node) is to consider users they have no lock-file ( https://github.com/actions/setup-node/pull/103 ).

**Comment by koba04** [comment]
@sosukesuzuki Thanks! That makes sense.

**Pull request merged**